Mathlib/RingTheory/Support.lean

```diff
@@ -0,0 +1,227 @@
+/-
+Copyright (c) 2024 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.Algebra.Module.LocalizedModule
+import Mathlib.AlgebraicGeometry.PrimeSpectrum.Basic
```

Comment:
What results here actually require the Zariski topology?

Reply:
That the support is closed in the zariski topology.

Reply:
`AlgebraicGeometry.PrimeSpectrum.Basic` was recently split into `RingTheory.PrimeSpectrum` (no topology) and `AlgebraicGeometry.PrimeSpectrum.Basic` (topology). We were getting some crazy dependencies otherwise. I think it would be good to the same here. This is a pretty basic commutative algebraic concept.

Reply:
Some part of it relies on `PrimeSpectrum.comap` and `PrimeSpectrum.basicOpen` that is in `AlgebraicGeometry.PrimeSpectrum.Basic` though. I've moved those results out into a separate file for now but do you think it makes sense to have unbundled versions of `PrimeSpectrum.comap` and `PrimeSpectrum.basicOpen` that does not depend on the topology?

Reply:
I think it makes sense but can be done later.